pykafka/pykafka/cluster.py

```diff
@@ -3,16 +3,38 @@
 import logging
 import time
 import random
+import weakref
 
 from .broker import Broker
 from .topic import Topic
 from .protocol import ConsumerMetadataRequest, ConsumerMetadataResponse
-from pykafka.exceptions import ConsumerCoordinatorNotAvailable
+from pykafka.exceptions import (ConsumerCoordinatorNotAvailable,
+                                UnknownTopicOrPartition)
 
 
 logger = logging.getLogger(__name__)
 
 
+class TopicDict(dict):
+    """Dictionary which will attempt to auto-create unknown topics."""
+
+    def __init__(self, cluster, *args, **kwargs):
+        super(TopicDict, self).__init__(*args, **kwargs)
+        self._cluster = weakref.proxy(cluster)
+
+    def __missing__(self, key):
+        logger.info('Topic %s not found. Attempting to auto-create.', key)
+        # Auto-creating will take a moment, so we try 3 times.
+        for i in xrange(3):
+            self._cluster.brokers[0].request_metadata(topics=[key])
```

Comment:
I'm not sure what this does in the context of creating a new topic, but it seems important. Maybe it should be rolled up into a convenience method along with the next line.